contracts/src/SuccinctGateway.sol

```diff
@@ -282,16 +298,32 @@ contract SuccinctGateway is ISuccinctGateway, FunctionRegistry, TimelockedUpgrad
 
     /// @notice Add the specified prover.
     /// @param _prover The address of the prover to add.
+    function addCustomProver(bytes32 functionId, address _prover) external {
+        if (msg.sender != verifierOwners[functionId]) {
+            revert NotFunctionOwner(msg.sender, verifierOwners[functionId]);
+        }
+        allowedProvers[functionId][_prover] = true;
+        emit ProverUpdated(functionId, _prover, true);
+    }
+
     function addProver(address _prover) external onlyGuardian {
-        allowedProvers[_prover] = true;
-        emit ProverUpdated(_prover, true);
+        allowedProvers[bytes32(0)][_prover] = true;
+        emit ProverUpdated(bytes32(0), _prover, true);
     }
 
     /// @notice Remove the specified prover.
     /// @param _prover The address of the prover to remove.
+    function removeCustomProver(bytes32 functionId, address _prover) external {
+        if (msg.sender != verifierOwners[functionId]) {
+            revert NotFunctionOwner(msg.sender, verifierOwners[functionId]);
+        }
+        allowedProvers[functionId][_prover] = false;
+        emit ProverUpdated(functionId, _prover, false);
+    }
+
     function removeProver(address _prover) external onlyGuardian {
```

Comment:
These functions could be renamed to addDefaultProver() and removeDefaultProver()